File: web/src/core/format/money.ts
// Spec: docs/specs/conventions.md (Money / números)

export const Money = new Intl.NumberFormat("pt-BR", {
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
});

export const Pct = new Intl.NumberFormat("pt-BR", {
  style: "percent",
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
});

export function formatMoney(v: number): string {
  return Money.format(v);
}

export function formatPct(v: number): string {
  return Pct.format(v);
}

/** "20k", "1,5k", ou "500" — usado em eixos de gráficos. */
export function moneyK(v: number): string {
  if (v == null || isNaN(v)) return "";
  if (Math.abs(v) >= 1000) {
    const k = v / 1000;
    return (k % 1 === 0 ? k.toString() : k.toFixed(1).replace(".", ",")) + "k";
  }
  return v.toLocaleString("pt-BR");
}

File: app/lib/core/format/money.dart
// Spec: docs/specs/conventions.md (Money / números)

import 'package:intl/intl.dart';

final NumberFormat _moneyFormatter = NumberFormat.decimalPattern('pt_BR')
  ..minimumFractionDigits = 2
  ..maximumFractionDigits = 2;

final NumberFormat _pctFormatter = NumberFormat.decimalPercentPattern(
  locale: 'pt_BR',
  decimalDigits: 2,
);

String formatMoney(double v) => _moneyFormatter.format(v);
String formatPct(double v) => _pctFormatter.format(v);

/// "20k", "1,5k", ou "500" — usado em eixos de gráficos.
String moneyK(num? v) {
  if (v == null || v.isNaN) return '';
  if (v.abs() >= 1000) {
    final k = v / 1000;
    if (k % 1 == 0) {
      return '${k.toInt()}k';
    }
    return '${k.toStringAsFixed(1).replaceAll('.', ',')}k';
  }
  return NumberFormat.decimalPattern('pt_BR').format(v);
}
